### ISSUE TITLE
Add a property to text rendering widgets that allows for adding a tooltip when the text is overflowed

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/flutter#116790]
In many cases for text widgets, we do not want to show a redundant tooltip (e.g. "This is my text widget with words" with a tooltip of "This is my text widget with words". But it would be useful to show a tooltip with the full content when the text is ellipsized (e.g "This is my text widg..." with a tooltip "This is my text widget with words". It would be useful if there was a `tooltipOnOverflow` property on `Text`(and likely on other text widgets with the `overflow` field).

**Comment by HansMuller** [comment]
CC @justinmc 
This sounds like a good idea

**Comment by justinmc** [comment]
I agree, we should make it easy to do this.  Maybe we could make this possible in a compositional way before considering adding a parameter to Text.  Something like this:

```dart
Tooltip(
  onTriggered: () {
    final Text text = textKey.currentWidget;
    if (text.isOverflowing) {
      return true;
    }
    return false;
  },
  child: Text(
    'very long text',
    key: textKey,
  ),
),
```

Though currently 1. onTriggered is just a VoidCallback, it doesn't decide whether to show the tooltip or not, and 2. there's nothing easy to determine if the text is overflowing like `isOverflowing`.

Also just as a side note, I thought that this happened automatically on the web (not Flutter), but it seems like I'm wrong, and you have to manually create a tooltip with CSS and/or JS.
